services/services.go

```diff
@@ -782,15 +823,55 @@ func getIndexPageData() (*types.IndexPageData, error) {
 		data.ActiveValidators = epochHistory[len(epochHistory)-1].ValidatorsCount
 	}
 
-	data.StakedEtherChartData = make([][]float64, len(epochHistory))
+	/* data.StakedEtherChartData = make([][]float64, len(epochHistory))
 	data.ActiveValidatorsChartData = make([][]float64, len(epochHistory))
 	for i, history := range epochHistory {
 		data.StakedEtherChartData[i] = []float64{float64(utils.EpochToTime(history.Epoch).Unix() * 1000), float64(history.EligibleEther) / 1000000000}
 		data.ActiveValidatorsChartData[i] = []float64{float64(utils.EpochToTime(history.Epoch).Unix() * 1000), float64(history.ValidatorsCount)}
-	}
+	} */
 
 	data.Subtitle = template.HTML(utils.Config.Frontend.SiteSubtitle)
+	data.ChurnRate = *GetLatestStats().ValidatorChurnLimit
+
+	err = db.ReaderDb.Get(&data.EthStore, `
+		SELECT apr
+		FROM eth_store_stats e
+		WHERE validator = -1 AND day = (SELECT MAX(day) FROM eth_store_stats);`)
+
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving eth store for index page: %v", err)
+	}
+
+	// get gas price history
+	now = time.Now().Truncate(time.Minute)
+	lastWeek := time.Now().Truncate(time.Minute).Add(-time.Hour * 24 * 5)
+
+	history, err := db.BigtableClient.GetGasNowHistory(now, lastWeek)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving gas price history: %v", err)
+	}
+
+	group := make(map[int64]float64, 0)
+	for i := 0; i < len(history); i++ {
+		_, ok := group[history[i].Ts.Truncate(time.Hour).Unix()]
+		if !ok {
+			group[history[i].Ts.Truncate(time.Hour).Unix()] = float64(history[i].Fast.Int64())
+		} else {
+			group[history[i].Ts.Truncate(time.Hour).Unix()] = (group[history[i].Ts.Truncate(time.Hour).Unix()] + float64(history[i].Fast.Int64())) / 2
```

Comment:
Is this formula correct? `group` works with hour precision and within `history` many entries for a specific hour may be present.
Let's say we have 3 entries with a gas price of 10, 5 and 1.
1) 10 gets saved as first value
2) (10 + 5) / 2 = 7.5 gets saved
3) (7.5 + 1) / 2 = **4.25** gets saved

But: (10 + 5 + 1) / 3 = **5,333333333**

Shouldn't 5,3 be the desired outcome?